src/pudl/io_managers.py

```diff
@@ -535,10 +508,29 @@ def load_input(self, context: InputContext) -> pd.DataFrame:
         return df
 
 
-@io_manager
-def pudl_sqlite_io_manager(init_context) -> PudlSQLiteIOManager:
+@io_manager(
+    config_schema={
+        "write_to_parquet": Field(
+            bool,
+            description="""If true, data will be written to parquet files,
+                in addition to the SQLite database.""",
+            default_value=bool(os.getenv("PUDL_WRITE_TO_PARQUET")),
+        ),
+        "read_from_parquet": Field(
+            bool,
+            description="""If True, the canonical source of data for reads
+                will be parquet files. Otherwise, data will be read from the
+                SQLite database.""",
+            default_value=bool(os.getenv("PUDL_READ_FROM_PARQUET")),
+        ),
+    }
+)
```

Comment:
Do we want to parameterize this using environment variables? I'm mildly against it because we don't use env vars to parameterize our other dagster resources.

Reply:
I think it makes sense to keep parquet support turned off by default for local development, so I generally agree with @rousik's reasoning that using environment variables will make it easy to turn on in a nightly build context. I do imagine this could be removed if/when parquet support gets past the experimental phase and we want to turn it on by default, but until then I personally think this would be useful.

Reply:
I think if it's turned off by default it'll see much less exercise and we may not catch issues as quickly. The entire PUDL DB is only ~1.3GB in Parquet. If it speeds up the ETL (or later, the full integration tests, or data validations) significantly, It'd be great to have them around.

Reply:
I'm in favor of turning on parquet outputs by default for the reasons Zane listed.

If we wanted to be consistent with how we are configuring the nightly build jobs right now we wouldn't use env vars and instead create a new argument for the `pudl_etl` cli called `--write-parquet-outputs` and pass the value to the `run_config` variable in `pudl.etl.cli`: https://github.com/catalyst-cooperative/pudl/blob/aaa46cbf98be6bf94303015eb358c5234837d3f1/src/pudl/etl/cli.py#L149-L158

I'm worried we'll end up with a confusing mix of methods for configuring runs: env vars, pudl_etl args and hard coded config in python code. I think eventually we should deprecate `pudl_etl` and create a preconfigued job in python for nightly build runs that can be executed using `dagster job execute nightly_build_job`. Something like this:

```python
defs: Definitions = Definitions(
    assets=default_assets,
    resources=default_resources,
    jobs=[
        define_asset_job(
            name="etl_full_nightly_build",
            config=default_config
            | {
                "resources": {
                  "datastore": {
                      "config": {
                          "gcs_cache_path": gcs_cache_path,
                      },
                  },
                 "pudl_io_manager": {
                     "config": {
                          "write_to_parquet": True,
                          "read_from_parquet": True
                     },
                  },
              },
            },
            description="This job the full ETL in a cloud VM.",
        ),
      # Other types of jobs: etl_fast, etl_fast_no_cems ...
    ],
)
```

Reply:
I'm fully convinced into parquet on-by-default, and not using environment variables, so I've up gone that direction.

@bendnorman it would definitely be nice to have all of the configuration for nightly builds consolidated in one place.

Reply:
I think switching to using the dagster CLI and their job configuration system directly sounds like a great medium term simplification.